src/state/positioner.ts

```diff
@@ -0,0 +1,210 @@
+import { uniqBy } from "lodash-es";
+import {
+  AnimalCard,
+  Card,
+  Coordinate,
+  DisasterCard,
+  ElementCard,
+  GameState,
+  PlantCard,
+  GamePieceTransforms,
+  UiState,
+} from "./types";
+import {
+  cardXOffset,
+  cardYOffset,
+  lowerXBoundary,
+  marketXStart,
+  marketYStart,
+  playerCardsYStart,
+  upperXBoundary,
+  upperYBoundary,
+} from "@/constants/gameBoard";
+import { cardHeight } from "@/constants/card";
+
+export const toUiState = (prevUiState: UiState | null, gameState: GameState): UiState => {
+  return {
+    cardPositions: calculateCardPositions(prevUiState, gameState),
+    deckPositions: calculateDeckPositions(gameState),
+  };
+};
+
+const calculateCardPositions = (prevUiState: UiState | null, gameState: GameState): GamePieceTransforms => {
+  return {
+    ...drawAnimalCards(prevUiState, gameState),
```

Comment:
Minor. Why `draw...`? maybe position/set/put/lay

Reply:
Yeah good point I called it draw initially and just kept the name, will think of something better